source/backend/api/Services/DispositionFileService.cs

```diff
@@ -246,6 +246,45 @@ public PimsDispositionSale GetDispositionFileSale(long dispositionFileId)
             return _dispositionFileRepository.GetDispositionFileSale(dispositionFileId);
         }
 
+        public PimsDispositionSale UpdateDispositionFileSale(long dispositionFileId, long saleId, PimsDispositionSale dispositionSale)
+        {
+            _logger.LogInformation("Updating disposition file Sale with DispositionFileId: {id}", dispositionFileId);
+            _user.ThrowIfNotAuthorized(Permissions.DispositionEdit);
+
+            var dispositionFileParent = _dispositionFileRepository.GetById(dispositionFileId);
+            if (dispositionFileId != dispositionSale.DispositionFileId || dispositionSale.DispositionSaleId != saleId || dispositionFileParent is null)
+            {
+                throw new BadRequestException("Invalid dispositionFileId.");
+            }
```

Comment:
this validation looks like it should happen on the controller instead of the service

Reply:
Controllers should not do validations, and be as simple as possible.

Reply:
that is not true. Please move api validation to the controller

Reply:
Right, the service probably only needs to know about the dispositionSale. If the service is always expecting the dispositionSale (which contains the saleId and dispositionFileId) to always match - we shouldn't need to pass those to the service at all.

Reply:
Another way to think of this is using Single Responsibility Principle. The only reason we have all of these parameters in the first place is because the controller needs them to be structured in a consistent manner for REST. Therefore, if these parameters are only **necessary** in the controller, they should not be passed onto other areas of the system.

Reply:
Updated